twilight-model/src/channel/message/component/select_menu.rs

```diff
@@ -1,27 +1,59 @@
-use crate::channel::message::ReactionType;
+use crate::channel::{message::ReactionType, ChannelType};
 use serde::{Deserialize, Serialize};
 
-/// Dropdown-style [`Component`] that renders belew messages.
+/// Dropdown-style [`Component`] that renders below messages.
 ///
 /// [`Component`]: super::Component
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct SelectMenu {
+    /// An optional list of channel types.
+    ///
+    /// This option is only used for [channel select menus](SelectMenuType::Channel).
+    pub channel_types: Option<Vec<ChannelType>>,
     /// Developer defined identifier.
     pub custom_id: String,
     /// Whether the select menu is disabled.
     ///
     /// Defaults to `false`.
     pub disabled: bool,
+    /// This select menu's type.
+    ///
+    /// [Text select menus](SelectMenuType::Text) *must* also set the `options` field.
```

Comment:
I'm not a particular fan of the language here and/or mentioning this in this place at all but I'll leave that for others to decide

Reply:
Considering it's already mentioned in the `option` field's documentation, I agree this is unnecessary.

I removed it in ab68bfb6.